
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Data Preparation &#8212; HiGlass v1.0 documentation</title>
    <link rel="stylesheet" href="_static/higlass.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     'v1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Track Types" href="track_types.html" />
    <link rel="prev" title="Javascript API" href="javascript_api.html" />
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-77756807-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-77756807-1');
  </script>
  
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />


  </head>
  <body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="data-preparation">
<h1>Data Preparation<a class="headerlink" href="#data-preparation" title="Permalink to this headline">¶</a></h1>
<p>Displaying large amounts of data often requires first turning it into
not-so-large amounts of data. Clodius is a program and library designed
to aggregate large datasets to make them easy to display at different
resolutions.</p>
<div class="section" id="bed-files">
<h2>Bed Files<a class="headerlink" href="#bed-files" title="Permalink to this headline">¶</a></h2>
<p>BED files specify genomic intervals. They are aggregated according to an
importance function that determines which values should be visible at lower
zoom levels. This importance function is user specified. In the absence of
any clear ranking of the different lines in the BED file, a random value
can be used in lieu of the importance function.</p>
<p>Example BED file:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>chr9    <span class="m">135766734</span>       <span class="m">135820020</span>       TSC1    Biallelic inactivation may predict sensitivity to MTOR inhibitors
chr16   <span class="m">2097895</span> <span class="m">2138721</span> TSC2    Biallelic inactivation may predict sensitivity to MTOR inhibitors
chr3    <span class="m">10183318</span>        <span class="m">10195354</span>        VHL     May signal the presence of a germline mutation.
chr11   <span class="m">32409321</span>        <span class="m">32457081</span>        WT1     May signal the presence of a germline mutation.
</pre></div>
</div>
<p>This file can be aggregated like so:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>clodius aggregate bedfile <span class="se">\</span>
    --assembly hg19 <span class="se">\</span>
    short.bed
</pre></div>
</div>
<p>And then imported into higlass after copying to the docker temp directory (<code class="docutils literal"><span class="pre">cp</span> <span class="pre">short.bed.multires</span> <span class="pre">~/hg-tmp/</span></code>):</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>docker <span class="nb">exec</span> higlass-container python <span class="se">\</span>
   higlass-server/manage.py ingest_tileset <span class="se">\</span>
       --filename /tmp/short.bed.multires <span class="se">\</span>
       --filetype beddb <span class="se">\</span>
       --datatype bedlike <span class="se">\</span>
       --coordSystem b37
</pre></div>
</div>
</div>
<div class="section" id="bedpe-like-files">
<h2>Bedpe-like Files<a class="headerlink" href="#bedpe-like-files" title="Permalink to this headline">¶</a></h2>
<p>BEDPE-like files contain two sets of chromosomal coordinates:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>chr10   <span class="m">74160000</span>        <span class="m">74720000</span>    chr10   <span class="m">74165000</span>    <span class="m">74725000</span>
chr12   <span class="m">120920000</span>       <span class="m">121640000</span>   chr12   <span class="m">120925000</span>   <span class="m">121645000</span>
chr15   <span class="m">86360000</span>        <span class="m">88840000</span>    chr15   <span class="m">86365000</span>    <span class="m">88845000</span>
</pre></div>
</div>
<p>To view such files in HiGlass, they have to be aggregated so that tiles don’t
contain too many values and slow down the renderer:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>clodius aggregate bedpe <span class="se">\</span>
    --assembly hg19 <span class="se">\</span>
    --chr1-col <span class="m">1</span> --from1-col <span class="m">2</span> --to1-col <span class="m">3</span> <span class="se">\</span>
    --chr2-col <span class="m">4</span> --from2-col <span class="m">5</span> --to2-col <span class="m">6</span> <span class="se">\</span>
    --output-file domains.txt.multires <span class="se">\</span>
    domains.txt
</pre></div>
</div>
<p>This requires the <cite>–chr1-col</cite>, <cite>–from1-col</cite>, <cite>–to1-col</cite>, <cite>–chr2-col</cite>,
<cite>–from2-col</cite>, <cite>–to2-col</cite> parameters to specify which columns in the datafile
describe the x-extent and y-extent of the region.</p>
<p>The priority with which regions are included in lower resolution tiles is
specified by the <cite>–impotance-column</cite> parameter. This can either provide a
value, contain <cite>random</cite>, or if it’s not specified, default to the size of the
region.</p>
</div>
<div class="section" id="bedgraph-files">
<h2>BedGraph files<a class="headerlink" href="#bedgraph-files" title="Permalink to this headline">¶</a></h2>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">The order of the chromosomes in the bedgraph file have to
be consistent with the order specified for the assembly in
<a class="reference external" href="https://github.com/pkerpedjiev/negspy/tree/master/negspy/data">the negspy repository</a>.</p>
</div>
<div class="section" id="ordering-the-chromosomes-in-the-input-file">
<h3>Ordering the chromosomes in the input file<a class="headerlink" href="#ordering-the-chromosomes-in-the-input-file" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="nv">input_file</span><span class="o">=</span>~/Downloads/phastCons100way.txt.gz<span class="p">;</span>
<span class="nv">output_file</span><span class="o">=</span>~/Downloads/phastConst100way_ordered.txt<span class="p">;</span>
<span class="nv">chromnames</span><span class="o">=</span><span class="k">$(</span>awk <span class="s1">&#39;{print $1}&#39;</span> ~/projects/negspy/negspy/data/hg19/chromInfo.txt<span class="k">)</span><span class="p">;</span>
<span class="k">for</span> chr in <span class="nv">$chromnames</span><span class="p">;</span>
    <span class="k">do</span> <span class="nb">echo</span> <span class="si">${</span><span class="nv">chr</span><span class="si">}</span><span class="p">;</span>
    zcat <span class="nv">$input_file</span> <span class="p">|</span> grep <span class="s2">&quot;\t</span><span class="si">${</span><span class="nv">chr</span><span class="si">}</span><span class="s2">\t&quot;</span> &gt;&gt; <span class="nv">$output_file</span><span class="p">;</span>
<span class="k">done</span><span class="p">;</span>
</pre></div>
</div>
</div>
<div class="section" id="aggregation-by-addition">
<h3>Aggregation by addition<a class="headerlink" href="#aggregation-by-addition" title="Permalink to this headline">¶</a></h3>
<p>Assume we have an input file that has <code class="docutils literal"><span class="pre">id</span> <span class="pre">chr</span> <span class="pre">start</span> <span class="pre">end</span> <span class="pre">value1</span> <span class="pre">value2</span></code> pairs:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">location</span>        <span class="n">chrom</span>   <span class="n">start</span>   <span class="n">end</span>     <span class="n">copynumber</span>      <span class="n">segmented</span>
<span class="mi">1</span><span class="p">:</span><span class="mi">2900001</span><span class="o">-</span><span class="mi">3000000</span>       <span class="mi">1</span>       <span class="mi">2900001</span> <span class="mi">3000000</span> <span class="o">-</span><span class="mf">0.614</span>  <span class="o">-</span><span class="mf">0.495</span>
<span class="mi">1</span><span class="p">:</span><span class="mi">3000001</span><span class="o">-</span><span class="mi">3100000</span>       <span class="mi">1</span>       <span class="mi">3000001</span> <span class="mi">3100000</span> <span class="o">-</span><span class="mf">0.407</span>  <span class="o">-</span><span class="mf">0.495</span>
<span class="mi">1</span><span class="p">:</span><span class="mi">3100001</span><span class="o">-</span><span class="mi">3200000</span>       <span class="mi">1</span>       <span class="mi">3100001</span> <span class="mi">3200000</span> <span class="o">-</span><span class="mf">0.428</span>  <span class="o">-</span><span class="mf">0.495</span>
<span class="mi">1</span><span class="p">:</span><span class="mi">3200001</span><span class="o">-</span><span class="mi">3300000</span>       <span class="mi">1</span>       <span class="mi">3200001</span> <span class="mi">3300000</span> <span class="o">-</span><span class="mf">0.437</span>  <span class="o">-</span><span class="mf">0.495</span>
</pre></div>
</div>
<p>We can aggregate this file by recursively summing adjacent values. We have to
indicate which column corresponds to the chromosome (<code class="docutils literal"><span class="pre">--chromosome-col</span> <span class="pre">2</span></code>),
the start position (<code class="docutils literal"><span class="pre">--from-pos-col</span> <span class="pre">3</span></code>), the end position (<code class="docutils literal"><span class="pre">--to-pos-col</span> <span class="pre">4</span></code>)
and the value column (<code class="docutils literal"><span class="pre">--value-col</span> <span class="pre">5</span></code>). We specify that the first line
of the data file contains a header using the (<code class="docutils literal"><span class="pre">--has-header</span></code>) option.</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>clodius aggregate bedgraph          <span class="se">\</span>
    test/sample_data/cnvs_hw.tsv    <span class="se">\</span>
    --output-file ~/tmp/cnvs_hw.hitile <span class="se">\</span>
    --chromosome-col <span class="m">2</span>              <span class="se">\</span>
    --from-pos-col <span class="m">3</span>                <span class="se">\</span>
    --to-pos-col <span class="m">4</span>                  <span class="se">\</span>
    --value-col <span class="m">5</span>                   <span class="se">\</span>
    --assembly grch37               <span class="se">\</span>
    --nan-value NA                  <span class="se">\</span>
    --transform exp2                <span class="se">\</span>
    --has-header
</pre></div>
</div>
<div class="section" id="data-transform">
<h4>Data Transform<a class="headerlink" href="#data-transform" title="Permalink to this headline">¶</a></h4>
<p>The dataset used in this example contains copy number data that has been log2
transformed. That is, the copy number given for each bin is the log2 of the
computed value. This is a problem for HiGlass’s default aggregation method of
summing adjacent values since <img class="math" src="_images/math/4e9125b9e00c5072d6f3577f7c356a015ed9ca82.png" alt="\log_2 a + \log_2 b \neq \log_2 ab"/>.</p>
<p>Using the <code class="docutils literal"><span class="pre">--transform</span> <span class="pre">exp2</span></code> option tells clodius to raise two to the
power of the provided value before doing the transformation and storing. As
an added benefit, NaN values become apparent in the resulting because they
have values of 0.</p>
</div>
<div class="section" id="nan-value-identification">
<h4>NaN Value Identification<a class="headerlink" href="#nan-value-identification" title="Permalink to this headline">¶</a></h4>
<p>NaN (not a number) values in the input file can be specified using the
<code class="docutils literal"><span class="pre">--nan-value</span></code> option.  For example, <code class="docutils literal"><span class="pre">--nan-value</span> <span class="pre">NA</span></code> indicates that
whenever <em>NA</em> is encountered as a value it should be treated as NaN. In the
current implementation, NaN values are simply treated as 0. In the future, they
should be assigned a special value so that they are ignored by <a class="reference external" href="http://higlass.io">HiGlass</a>.</p>
<p>When NaN values are aggregated by summing, they are treated as 0 when added to
another number. When two NaN values are added to each other, however, the
result is Nan.</p>
</div>
<div class="section" id="nan-value-counting">
<h4>NaN Value Counting<a class="headerlink" href="#nan-value-counting" title="Permalink to this headline">¶</a></h4>
<p>Sometimes, we just want to count the number of NaN values in the file. The
<code class="docutils literal"><span class="pre">--count-nan</span></code> option effectively treats NaN values as 1 and all other values
as 0. This makes it possible to display a track showing how many NaN values are
present in each interval. It also makes it possible to create compound tracks
which use that information to normalize track values.</p>
</div>
</div>
</div>
<div class="section" id="bigwig-files">
<h2>bigWig files<a class="headerlink" href="#bigwig-files" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://genome.ucsc.edu/goldenpath/help/bigWig.html">bigWig files</a> store
genomic data in a compressed, indexed form that allows rapid retrieval and
visualization. bigWig files can be loaded directly into HiGlass using the
vector datatype and bigwig filetype:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>docker <span class="nb">exec</span> higlass-container python <span class="se">\</span>
        higlass-server/manage.py ingest_tileset <span class="se">\</span>
        --filename /tmp/cnvs_hw.hitile <span class="se">\</span>
        --filetype bigwig <span class="se">\</span>
        --datatype vector
</pre></div>
</div>
</div>
<div class="section" id="gene-annotation-tracks">
<h2>Gene Annotation Tracks<a class="headerlink" href="#gene-annotation-tracks" title="Permalink to this headline">¶</a></h2>
<p>HiGlass uses a specialized track for displaying gene annotations. It is rougly
based on UCSC’s refGene files
(e.g. <a class="reference external" href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/</a>). For any identifiable
genome assembly the following commands can be run to generate a list of
gene annotation that can be loaded as a zoomable track in HiGlass.</p>
<div class="section" id="prerequisites">
<h3>Prerequisites<a class="headerlink" href="#prerequisites" title="Permalink to this headline">¶</a></h3>
<p>For any assembly, there needs to a refGene file:</p>
<p><a class="reference external" href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/refGene.txt.gz">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/refGene.txt.gz</a></p>
<p>And a list of chromosome sizes in the <a class="reference external" href="https://github.com/pkerpedjiev/negspy">negspy</a> python package.</p>
<p>If there are no available chromosome sizes for this assembly in negspy, adding
them is simply a matter of downloading the list from UCSC (e.g.
<a class="reference external" href="http://hgdownload.cse.ucsc.edu/goldenpath/hg19/bigZips/hg19.chrom.sizes">http://hgdownload.cse.ucsc.edu/goldenpath/hg19/bigZips/hg19.chrom.sizes</a>)</p>
</div>
<div class="section" id="set-the-assembly-name-and-species-id">
<h3>Set the assembly name and species ID<a class="headerlink" href="#set-the-assembly-name-and-species-id" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="nv">ASSEMBLY</span><span class="o">=</span>mm9
<span class="nv">TAXID</span><span class="o">=</span><span class="m">10090</span>

<span class="c1">#ASSEMBLY=hg19</span>
<span class="c1">#TAXID=9606</span>

<span class="c1">#ASSEMBLY=sacCer3</span>
<span class="c1">#TAXID=559292</span>

<span class="c1">#ASSEMBLY=dm6</span>
<span class="c1">#TAXID=7227</span>
</pre></div>
</div>
</div>
<div class="section" id="download-data-from-ucsc-and-ncbi">
<h3>Download data from UCSC and NCBI<a class="headerlink" href="#download-data-from-ucsc-and-ncbi" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash"><div class="highlight"><pre><span></span>mkdir -p ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>

wget -N -P ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/ <span class="se">\</span>
    http://hgdownload.cse.ucsc.edu/goldenPath/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/database/refGene.txt.gz

wget -N -P ~/data/genbank-data/ <span class="se">\</span>
    ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/gene2refseq.gz

wget -N -P ~/data/genbank-data/ <span class="se">\</span>
    ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/gene_info.gz

wget -N -P ~/data/genbank-data/ <span class="se">\</span>
    ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/gene2pubmed.gz
</pre></div>
</div>
</div>
<div class="section" id="preprocess-data">
<h3>Preprocess data<a class="headerlink" href="#preprocess-data" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="c1"># remove entries to chr6_...</span>

gzcat ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/refGene.txt.gz <span class="se">\</span>
    <span class="p">|</span> awk -F <span class="s1">$&#39;\t&#39;</span> <span class="s1">&#39;{if (!($3 ~ /_/)) print;}&#39;</span> <span class="se">\</span>
    <span class="p">|</span> sort -k2 &gt; ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/sorted_refGene
wc -l ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/sorted_refGene

zgrep ^<span class="si">${</span><span class="nv">TAXID</span><span class="si">}</span> ~/data/genbank-data/gene2refseq.gz <span class="se">\</span>
     &gt; ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene2refseq
head ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene2refseq

zgrep ^<span class="si">${</span><span class="nv">TAXID</span><span class="si">}</span> ~/data/genbank-data/gene_info.gz <span class="se">\</span>
    <span class="p">|</span> sort -k <span class="m">2</span> <span class="se">\</span>
     &gt; ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene_info
head ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene_info

zgrep ^<span class="si">${</span><span class="nv">TAXID</span><span class="si">}</span> ~/data/genbank-data/gene2pubmed.gz <span class="se">\</span>
    &gt; ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene2pubmed
head ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene2pubmed

<span class="c1"># awk &#39;{print $2}&#39; ~/data/genbank-data/hg19/gene_info \</span>
<span class="c1"># | xargs python scripts/gene_info_by_id.py \</span>
<span class="c1"># | tee ~/data/genbank-data/hg19/gene_summaries.tsv</span>

<span class="c1"># output -&gt; geneid \t citation_count</span>
</pre></div>
</div>
</div>
<div class="section" id="processing">
<h3>Processing<a class="headerlink" href="#processing" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash"><div class="highlight"><pre><span></span>cat ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene2pubmed <span class="se">\</span>
    <span class="p">|</span> awk <span class="s1">&#39;{print $2}&#39;</span> <span class="se">\</span>
    <span class="p">|</span> sort <span class="se">\</span>
    <span class="p">|</span> uniq -c <span class="se">\</span>
    <span class="p">|</span> awk <span class="s1">&#39;{print $2 &quot;\t&quot; $1}&#39;</span> <span class="se">\</span>
    <span class="p">|</span> sort <span class="se">\</span>
    &gt; ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene2pubmed-count
head ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene2pubmed-count


<span class="c1"># output -&gt; geneid \t refseq_id</span>

cat ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene2refseq <span class="se">\</span>
    <span class="p">|</span> awk -F <span class="s1">$&#39;\t&#39;</span> <span class="s1">&#39;{ split($4,a,&quot;.&quot;); if (a[1] != &quot;-&quot;) print $2 &quot;\t&quot; a[1];}&#39;</span> <span class="se">\</span>
    <span class="p">|</span> sort <span class="se">\</span>
    <span class="p">|</span> uniq  <span class="se">\</span>
    &gt; ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneid_refseqid
head ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneid_refseqid
wc -l ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneid_refseqid


<span class="c1">#output -&gt; geneid \t refseq_id \t citation_count</span>

join ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneid_refseqid <span class="se">\</span>
    ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene2pubmed-count  <span class="se">\</span>
    <span class="p">|</span> sort -k2 <span class="se">\</span>
    &gt; ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneid_refseqid_count

head ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneid_refseqid_count
wc -l ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneid_refseqid_count


<span class="c1"># output -&gt; geneid \t refseq_id \t chr (5) \t strand(6) \t txStart(7) \t txEnd(8) \t cdsStart(9) \t cdsEnd (10) \t exonCount(11) \t exonStarts(12) \t exonEnds(13)</span>

join -1 <span class="m">2</span> -2 <span class="m">2</span> <span class="se">\</span>
    ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneid_refseqid_count <span class="se">\</span>
    ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/sorted_refGene <span class="se">\</span>
    <span class="p">|</span> awk <span class="s1">&#39;{ print $2 &quot;\t&quot; $1 &quot;\t&quot; $5 &quot;\t&quot; $6 &quot;\t&quot; $7 &quot;\t&quot; $8 &quot;\t&quot; $9 &quot;\t&quot; $10 &quot;\t&quot; $11 &quot;\t&quot; $12 &quot;\t&quot; $13 &quot;\t&quot; $3; }&#39;</span> <span class="se">\</span>
    <span class="p">|</span> sort -k1   <span class="se">\</span>
    &gt; ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneid_refGene_count

head ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneid_refGene_count
wc -l ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneid_refGene_count

<span class="c1"># output -&gt; geneid \t symbol \t gene_type \t name \t citation_count</span>

join -1 <span class="m">2</span> -2 <span class="m">1</span> -t <span class="s1">$&#39;\t&#39;</span> <span class="se">\</span>
    ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene_info <span class="se">\</span>
    ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene2pubmed-count <span class="se">\</span>
    <span class="p">|</span> awk -F <span class="s1">$&#39;\t&#39;</span> <span class="s1">&#39;{print $1 &quot;\t&quot; $3 &quot;\t&quot; $10 &quot;\t&quot; $12 &quot;\t&quot; $16}&#39;</span> <span class="se">\</span>
    <span class="p">|</span> sort -k1 <span class="se">\</span>
    &gt; ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene_subinfo_citation_count
head ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene_subinfo_citation_count
wc -l ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene_subinfo_citation_count


<span class="c1"># 1: chr (chr1)</span>
<span class="c1"># 2: txStart (52301201) [9]</span>
<span class="c1"># 3: txEnd (52317145) [10]</span>
<span class="c1"># 4: geneName (ACVRL1)   [2]</span>
<span class="c1"># 5: citationCount (123) [16]</span>
<span class="c1"># 6: strand (+)  [8]</span>
<span class="c1"># 7: refseqId (NM_000020)</span>
<span class="c1"># 8: geneId (94) [1]</span>
<span class="c1"># 9: geneType (protein-coding)</span>
<span class="c1"># 10: geneDesc (activin A receptor type II-like 1)</span>
<span class="c1"># 11: cdsStart (52306258)</span>
<span class="c1"># 12: cdsEnd (52314677)</span>
<span class="c1"># 14: exonStarts (52301201,52306253,52306882,52307342,52307757,52308222,52309008,52309819,52312768,52314542,)</span>
<span class="c1"># 15: exonEnds (52301479,52306319,52307134,52307554,52307857,52308369,52309284,52310017,52312899,52317145,)</span>

join -t <span class="s1">$&#39;\t&#39;</span> <span class="se">\</span>
    ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/gene_subinfo_citation_count <span class="se">\</span>
    ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneid_refGene_count <span class="se">\</span>
    <span class="p">|</span> awk -F <span class="s1">$&#39;\t&#39;</span> <span class="s1">&#39;{print $7 &quot;\t&quot; $9 &quot;\t&quot; $10 &quot;\t&quot; $2 &quot;\t&quot; $16 &quot;\t&quot; $8 &quot;\t&quot; $6 &quot;\t&quot; $1 &quot;\t&quot; $3 &quot;\t&quot; $4 &quot;\t&quot; $11 &quot;\t&quot; $12 &quot;\t&quot; $14 &quot;\t&quot; $15}&#39;</span> <span class="se">\</span>
    &gt; ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneAnnotations.bed
head ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneAnnotations.bed
wc -l ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneAnnotations.bed

python scripts/exonU.py <span class="se">\</span>
    ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneAnnotations.bed <span class="se">\</span>
    &gt; ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneAnnotationsExonUnions.bed
wc -l ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneAnnotationsExonUnions.bed
</pre></div>
</div>
</div>
<div class="section" id="creating-a-higlass-track">
<h3>Creating a HiGlass Track<a class="headerlink" href="#creating-a-higlass-track" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash"><div class="highlight"><pre><span></span>workon hg-server
<span class="nv">ASSEMBLY</span><span class="o">=</span>mm9

clodius aggregate bedfile <span class="se">\</span>
    --max-per-tile <span class="m">20</span> --importance-column <span class="m">5</span> <span class="se">\</span>
    --assembly <span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span> <span class="se">\</span>
    --output-file ~/data/tiled-data/gene-annotations-<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>.db <span class="se">\</span>
    --delimiter <span class="s1">$&#39;\t&#39;</span> <span class="se">\</span>
    ~/data/genbank-data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/geneAnnotationsExonUnions.bed

aws s3 cp ~/data/tiled-data/gene-annotations-<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>.db <span class="se">\</span>
    s3://pkerp/public/hg-server/data/<span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span>/
</pre></div>
</div>
</div>
<div class="section" id="importing-into-higlass">
<h3>Importing into HiGlass<a class="headerlink" href="#importing-into-higlass" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash"><div class="highlight"><pre><span></span>curl -u <span class="sb">`</span>cat ~/.higlass-server-login<span class="sb">`</span>    <span class="se">\</span>
    -F <span class="s2">&quot;datafile=@/Users/peter/data/tiled-data/gene-annotations-</span><span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span><span class="s2">.db&quot;</span>    <span class="se">\</span>
    -F <span class="s2">&quot;name=Gene Annotations (</span><span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span><span class="s2">)&quot;</span>   <span class="se">\</span>
    -F <span class="s1">&#39;filetype=beddb&#39;</span>  <span class="se">\</span>
    -F <span class="s1">&#39;datatype=gene-annotation&#39;</span>  <span class="se">\</span>
    -F <span class="s1">&#39;coordSystem=${ASSEMBLY}&#39;</span> <span class="se">\</span>
    -F <span class="s1">&#39;coordSystem2=${ASSEMBLY}&#39;</span>  <span class="se">\</span>
    http://higlass.io:80/api/v1/tilesets/
</pre></div>
</div>
</div>
<div class="section" id="chromosomes">
<h3>Chromosomes<a class="headerlink" href="#chromosomes" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash"><div class="highlight"><pre><span></span>curl -u <span class="sb">`</span>cat ~/.higlass-server-login<span class="sb">`</span>    <span class="se">\</span>
    -F <span class="s2">&quot;datafile=@/Users/peter/tmp/chromSizes_hg38.tsv&quot;</span>    <span class="se">\</span>
    -F <span class="s2">&quot;name=Chromosomes (hg38)&quot;</span>   <span class="se">\</span>
    -F <span class="s1">&#39;filetype=chromsizes-tsv&#39;</span>  <span class="se">\</span>
    -F <span class="s1">&#39;datatype=chromsizes&#39;</span>  <span class="se">\</span>
    -F <span class="s2">&quot;coordSystem=</span><span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span><span class="s2">&quot;</span> <span class="se">\</span>
    -F <span class="s2">&quot;coordSystem2=</span><span class="si">${</span><span class="nv">ASSEMBLY</span><span class="si">}</span><span class="s2">&quot;</span>  <span class="se">\</span>
    http://higlass.io:80/api/v1/tilesets/
</pre></div>
</div>
</div>
</div>
<div class="section" id="hitile-files">
<h2>Hitile files<a class="headerlink" href="#hitile-files" title="Permalink to this headline">¶</a></h2>
<p>Hitile files are HDF5-based 1D vector files containing data at multiple resolutions.</p>
<p>To see hitile datasets in higlass, use the docker container to load them:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>docker <span class="nb">exec</span> higlass-container python <span class="se">\</span>
        higlass-server/manage.py ingest_tileset <span class="se">\</span>
        --filename /tmp/cnvs_hw.hitile <span class="se">\</span>
        --filetype hitile <span class="se">\</span>
        --datatype vector
</pre></div>
</div>
<p>Point your browser at 127.0.0.1:8989 (or wherever it is hosted), click on the
little ‘plus’ icon in the view and select the top position.  You will see a
listing of available tracks that can be loaded. Select the dataset and then
choose the plot type to display it as.</p>
</div>
<div class="section" id="cooler-files">
<h2>Cooler files<a class="headerlink" href="#cooler-files" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://github.com/mirnylab/cooler">Cooler files</a> (extension .cool) store
arbitrarily large 2D genomic matrices, such as those produced via Hi-C and other high
throughput proximity ligation experiments. HiGlass can render cooler files containing
matrices of the same dataset at a range of bin resolutions or <em>zoom levels</em>, so called multiresolution
cool files (typically denoted .mcool).</p>
<div class="section" id="from-pairs">
<h3>From pairs<a class="headerlink" href="#from-pairs" title="Permalink to this headline">¶</a></h3>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Starting with <em>cooler</em> 0.7.9, input pairs data no longer needs to be sorted and indexed.</p>
</div>
<p>Often you will start with a <strong>list of pairs</strong> (e.g. contacts, interactions) that need to be aggregated.
For example, the 4DN-DCIC developed a <a class="reference external" href="https://github.com/4dn-dcic/pairix/blob/master/pairs_format_specification.md">standard pairs format</a> for HiC-like data. In general, you
only need a tab-delimited file with columns representing <code class="docutils literal"><span class="pre">chrom1</span></code>, <code class="docutils literal"><span class="pre">pos1</span></code>, <code class="docutils literal"><span class="pre">chrom2</span></code>, <code class="docutils literal"><span class="pre">pos2</span></code>, optionally gzipped. In the case of Hi-C, these would correspond to the mapped locations of the two ends of a Hi-C ligation product.</p>
<p>You also need to provide a list of chromosomes in semantic order (chr1, chr2, …, chrX, chrY, …) in a
two-column <a class="reference external" href="https://github.com/pkerpedjiev/negspy/blob/master/negspy/data/hg19/chromSizes.tsv">chromsizes</a> file.</p>
<p>Ingesting pairs is done using the <code class="docutils literal"><span class="pre">cooler</span> <span class="pre">cload</span></code> command. Choose the appropriate loading subcommand. If you pairs file is sorted and indexed with <a class="reference external" href="https://github.com/4dn-dcic/pairix">pairix</a> or with <a class="reference external" href="https://davetang.org/muse/2013/02/22/using-tabix/">tabix</a>, use <code class="docutils literal"><span class="pre">cooler</span> <span class="pre">cload</span> <span class="pre">pairix</span></code> or <code class="docutils literal"><span class="pre">cooler</span> <span class="pre">cload</span> <span class="pre">tabix</span></code>, respectively. Otherwise, you can use the new <code class="docutils literal"><span class="pre">cooler</span> <span class="pre">cload</span> <span class="pre">pairs</span></code> command.</p>
<p><strong>Raw pairs example</strong></p>
<p>If you have a raw pairs file or you can stream your data in such a way, you only need to specify the columns that correspond to <cite>chrom1</cite>, <cite>chrom2</cite>, <cite>pos1</cite> and <cite>pos2</cite>. For example, if <code class="docutils literal"><span class="pre">chrom1</span></code> and <code class="docutils literal"><span class="pre">pos1</span></code> are the first two columns, and <code class="docutils literal"><span class="pre">chrom2</span></code> and <code class="docutils literal"><span class="pre">pos2</span></code> are in columns 4 and 5, the following command will aggregate the input pairs at 1kb:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>cooler cload pairs -c1 <span class="m">1</span> -p1 <span class="m">2</span> -c2 <span class="m">4</span> -p2 <span class="m">5</span> <span class="se">\</span>
    hg19.chrom.sizes:1000 <span class="se">\</span>
    mypairs.txt <span class="se">\</span>
    mycooler.1000.cool
</pre></div>
</div>
<p>To pipe in a stream, replace the pairs path above with a dash <code class="docutils literal"><span class="pre">-</span></code>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The syntax <code class="docutils literal"><span class="pre">&lt;chromsizes_path&gt;:&lt;binsize_in_bp&gt;</span></code> is a shortcut to specify the genomic bin segmentation used to aggregate the pairs. Alternatively, you can pass in the path to a 3-column BED file of bins.</p>
</div>
<p><strong>Indexed pairs example</strong></p>
<p>If you want to create a sorted and indexed pairs file, follow this example. Because an index provides random access to the pairs, this method can be more efficient and parallelized.</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>cooler csort -c1 <span class="m">1</span> -p1 <span class="m">2</span> -c2 <span class="m">4</span> -p2 <span class="m">5</span> mypairs.txt hg19.chrom.sizes
</pre></div>
</div>
<p>will generate a sorted and compressed pairs file <code class="docutils literal"><span class="pre">mypairs.blksrt.txt.gz</span></code> along with a companion pairix <code class="docutils literal"><span class="pre">.px2</span></code> index file. To aggregate, use the <code class="docutils literal"><span class="pre">cload</span> <span class="pre">pairix</span></code> command.</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>cooler cload pairix hg19.chrom.sizes:1000 mypairs.blksrt.txt.gz mycooler.1000.cool
</pre></div>
</div>
<p>The output <code class="docutils literal"><span class="pre">mycooler.1000.cool</span></code> will serve as the <em>base resolution</em> for the multires cooler you will generate.</p>
</div>
<div class="section" id="from-a-matrix">
<h3>From a matrix<a class="headerlink" href="#from-a-matrix" title="Permalink to this headline">¶</a></h3>
<p>If your base resolution data is <strong>already aggregated</strong>, you can ingest data in one of two formats. Use <code class="docutils literal"><span class="pre">cooler</span> <span class="pre">load</span></code> to ingest.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Prior to <em>cooler</em> 0.7.9, input BG2 files needed to be sorted and indexed. This is no longer the case.</p>
</div>
<ol class="arabic simple">
<li><strong>COO</strong>: Sparse matrix upper triangle <a class="reference external" href="https://en.wikipedia.org/wiki/Sparse_matrix#Coordinate_list_(COO)">coordinate list</a> , i.e. tab-delimited sparse matrix triples (<code class="docutils literal"><span class="pre">row_id</span></code>, <code class="docutils literal"><span class="pre">col_id</span></code>, <code class="docutils literal"><span class="pre">count</span></code>). This is an output of pipelines like HiCPro.</li>
</ol>
<div class="highlight-bash"><div class="highlight"><pre><span></span>cooler load -f coo hg19.chrom.sizes:1000 mymatrix.1kb.coo.txt mycooler.1000.cool
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><strong>BG2</strong>: A 2D “extension” of the <a class="reference external" href="https://genome.ucsc.edu/goldenpath/help/bedgraph.html">bedGraph</a> format. Tab delimited with columns representing <code class="docutils literal"><span class="pre">chrom1</span></code>, <code class="docutils literal"><span class="pre">start1</span></code>, <code class="docutils literal"><span class="pre">end1</span></code>, <code class="docutils literal"><span class="pre">chrom2</span></code>, <code class="docutils literal"><span class="pre">start2</span></code>, <code class="docutils literal"><span class="pre">end2</span></code>, and <code class="docutils literal"><span class="pre">count</span></code>.</li>
</ol>
<div class="highlight-bash"><div class="highlight"><pre><span></span>cooler load -f bg2 hg19.chrom.sizes:1000 mymatrix.1kb.bg2.gz mycooler.1000.cool
</pre></div>
</div>
</div>
<div class="section" id="zoomify">
<h3>Zoomify<a class="headerlink" href="#zoomify" title="Permalink to this headline">¶</a></h3>
<p>To recursively aggregate your matrix into a multires file, use the <code class="docutils literal"><span class="pre">zoomify</span></code> command.</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>cooler zoomify mycooler.1000.cool
</pre></div>
</div>
<p>The output will be a file called <code class="docutils literal"><span class="pre">mycooler.1000.mcool</span></code> with zoom levels increasing by factors of 2. You can also
request an explicit list of resolutions, as long as they can be obtained via integer multiples starting from the base resolution. HiGlass performs well as long as zoom levels don’t differ in resolution by greater than a factor of ~5.</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>cooler zoomify -r <span class="m">5000</span>,10000,25000,50000,100000,500000,1000000 mycooler.1000.cool
</pre></div>
</div>
<p>If this is Hi-C data or similar, you probably want to apply iterative correction (i.e. matrix balancing normalization) by including the <code class="docutils literal"><span class="pre">--balance</span></code> option.</p>
</div>
<div class="section" id="loading-pre-zoomed-data">
<h3>Loading pre-zoomed data<a class="headerlink" href="#loading-pre-zoomed-data" title="Permalink to this headline">¶</a></h3>
<p>If the matrices for the resolutions you wish to visualize are already available, you can ingest each one independently into the right location inside the file using the <a class="reference external" href="http://cooler.readthedocs.io/en/latest/api.html#uri-string">Cooler URI</a> <code class="docutils literal"><span class="pre">::</span></code> syntax.</p>
<p>HiGlass expects each zoom level to be stored at a location named <code class="docutils literal"><span class="pre">resolutions/{binsize}</span></code>.</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>cooler load -f bg2 hg19.chrom.sizes:1000 mymatrix.1kb.bg2 mycooler.mcool::resolutions/1000
cooler load -f bg2 hg19.chrom.sizes:5000 mymatrix.5kb.bg2 mycooler.mcool::resolutions/5000
cooler load -f bg2 hg19.chrom.sizes:10000 mymatrix.10kb.bg2 mycooler.mcool::resolutions/10000
...
</pre></div>
</div>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last">See the <em>cooler</em> <a class="reference external" href="http://cooler.readthedocs.io/">docs</a> for more information.
You can also type <code class="docutils literal"><span class="pre">-h</span></code> or <code class="docutils literal"><span class="pre">--help</span></code> after any cooler command for a detailed description.</p>
</div>
</div>
</div>
<div class="section" id="multivec-files">
<span id="loading-into-higlass"></span><h2>Multivec Files<a class="headerlink" href="#multivec-files" title="Permalink to this headline">¶</a></h2>
<p>Multivec files store arrays of arrays organized by chromosome. To aggregate this
data, we need an input file where chromsome is a separate dataset. Example:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="n">f</span> <span class="o">=</span> <span class="n">h5py</span><span class="o">.</span><span class="n">File</span><span class="p">(</span><span class="s1">&#39;/tmp/blah.h5&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span>

<span class="n">d</span> <span class="o">=</span> <span class="n">f</span><span class="o">.</span><span class="n">create_dataset</span><span class="p">(</span><span class="s1">&#39;chr1&#39;</span><span class="p">,</span> <span class="p">(</span><span class="mi">10000</span><span class="p">,</span><span class="mi">5</span><span class="p">),</span> <span class="n">compression</span><span class="o">=</span><span class="s1">&#39;gzip&#39;</span><span class="p">)</span>
<span class="n">d</span><span class="p">[:]</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">random</span><span class="p">((</span><span class="mi">10000</span><span class="p">,</span><span class="mi">5</span><span class="p">))</span>
<span class="n">f</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>
</pre></div>
</div>
<p>This can be aggregated to multiple resolutions using <cite>clodius aggregate multivec</cite>:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>clodius aggregate multivec <span class="se">\</span>
    --chromsizes-filename ~/projects/negspy/negspy/data/hg38/chromInfo.txt <span class="se">\</span>
    --starting-resolution <span class="m">1000</span> <span class="se">\</span>
    --row-infos-filename ~/Downloads/sampled_info.txt <span class="se">\</span>
    my_file_genome_wide_hg38_v2.multivec
</pre></div>
</div>
<p>The <cite>–chromsizes-filename</cite> option lists the chromosomes that are in the input
file and their sizes.  The <cite>–starting-resolution</cite> option indicates that the
base resolution for the input data is 1000 base pairs.</p>
</div>
<div class="section" id="epilogos-data-multivec">
<h2>Epilogos Data (multivec)<a class="headerlink" href="#epilogos-data-multivec" title="Permalink to this headline">¶</a></h2>
<p>Epilogos (<a class="reference external" href="https://epilogos.altiusinstitute.org/">https://epilogos.altiusinstitute.org/</a>) show the distribution of chromatin states
over a set of experimental conditions (e.g. cell lines). The data consist of positions and
states:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">chr1</span>    <span class="mi">10000</span>   <span class="mi">10200</span>   <span class="nb">id</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span><span class="n">qcat</span><span class="p">:[</span> <span class="p">[</span><span class="o">-</span><span class="mf">0.2833</span><span class="p">,</span><span class="mi">15</span><span class="p">],</span> <span class="p">[</span><span class="o">-</span><span class="mf">0.04748</span><span class="p">,</span><span class="mi">5</span><span class="p">],</span> <span class="p">[</span><span class="o">-</span><span class="mf">0.008465</span><span class="p">,</span><span class="mi">7</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">3</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">4</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">6</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">10</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">11</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">12</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">13</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">14</span><span class="p">],</span> <span class="p">[</span><span class="mf">0.0006647</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span> <span class="p">[</span><span class="mf">0.436</span><span class="p">,</span><span class="mi">8</span><span class="p">],</span> <span class="p">[</span><span class="mf">1.921</span><span class="p">,</span><span class="mi">9</span><span class="p">]</span> <span class="p">]</span>
<span class="n">chr1</span>    <span class="mi">10200</span>   <span class="mi">10400</span>   <span class="nb">id</span><span class="p">:</span><span class="mi">2</span><span class="p">,</span><span class="n">qcat</span><span class="p">:[</span> <span class="p">[</span><span class="o">-</span><span class="mf">0.2833</span><span class="p">,</span><span class="mi">15</span><span class="p">],</span> <span class="p">[</span><span class="o">-</span><span class="mf">0.04748</span><span class="p">,</span><span class="mi">5</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">3</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">4</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">6</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">7</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">10</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">11</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">12</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">13</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">14</span><span class="p">],</span> <span class="p">[</span><span class="mf">0.0006647</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span> <span class="p">[</span><span class="mf">0.004089</span><span class="p">,</span><span class="mi">2</span><span class="p">],</span> <span class="p">[</span><span class="mf">0.8141</span><span class="p">,</span><span class="mi">8</span><span class="p">],</span> <span class="p">[</span><span class="mf">1.706</span><span class="p">,</span><span class="mi">9</span><span class="p">]</span> <span class="p">]</span>
<span class="n">chr1</span>    <span class="mi">10400</span>   <span class="mi">10600</span>   <span class="nb">id</span><span class="p">:</span><span class="mi">3</span><span class="p">,</span><span class="n">qcat</span><span class="p">:[</span> <span class="p">[</span><span class="o">-</span><span class="mf">0.2588</span><span class="p">,</span><span class="mi">15</span><span class="p">],</span> <span class="p">[</span><span class="o">-</span><span class="mf">0.04063</span><span class="p">,</span><span class="mi">5</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">3</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">4</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">6</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">7</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">10</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">11</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">12</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">13</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">14</span><span class="p">],</span> <span class="p">[</span><span class="mf">0.0006647</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span> <span class="p">[</span><span class="mf">0.2881</span><span class="p">,</span><span class="mi">8</span><span class="p">],</span> <span class="p">[</span><span class="mf">1.58</span><span class="p">,</span><span class="mi">9</span><span class="p">]</span> <span class="p">]</span>
<span class="n">chr1</span>    <span class="mi">10600</span>   <span class="mi">10800</span>   <span class="nb">id</span><span class="p">:</span><span class="mi">4</span><span class="p">,</span><span class="n">qcat</span><span class="p">:[</span> <span class="p">[</span><span class="o">-</span><span class="mf">0.02619</span><span class="p">,</span><span class="mi">15</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">3</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">4</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">6</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">7</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">8</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">10</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">11</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">12</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">13</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">14</span><span class="p">],</span> <span class="p">[</span><span class="mf">0.1077</span><span class="p">,</span><span class="mi">5</span><span class="p">],</span> <span class="p">[</span><span class="mf">0.4857</span><span class="p">,</span><span class="mi">9</span><span class="p">]</span> <span class="p">]</span>
</pre></div>
</div>
<p>This can be aggregated into multivec format:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>clodius convert bedfile_to_multivec <span class="se">\</span>
    hg38/all.KL.bed.gz <span class="se">\</span>
    --assembly hg38 <span class="se">\</span>
    --starting-resolution <span class="m">200</span> <span class="se">\</span>
    --row-infos-filename row_infos.txt <span class="se">\</span>
    --num-rows <span class="m">15</span> <span class="se">\</span>
    --format epilogos
</pre></div>
</div>
</div>
<div class="section" id="other-data-multivec">
<h2>Other Data (multivec)<a class="headerlink" href="#other-data-multivec" title="Permalink to this headline">¶</a></h2>
<p>Multivec files are datatype agnostic. For use with generic data, create a
<cite>segments</cite> file containing the maximum value for each segment. A segment is an
arbitrary set of discontinuous blocks that the data is partitioned into. If the
data has no natural grouping, one segment can be used which contains the
maximum x value in the dataset:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>segment1    <span class="m">20000</span>
</pre></div>
</div>
<p>The individual datapoints should then be formatted as follows:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>segment_name    start   end value
segment1    <span class="m">10</span>  <span class="m">100</span> <span class="m">5</span>
</pre></div>
</div>
<p>This can be converted to a multivec file using the following command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>clodius convert bedfile_to_multivec <span class="se">\</span>
    data.tsv <span class="se">\</span>
    --chromsizes-file segments.tsv <span class="se">\</span>
    --starting-resolution <span class="m">1</span>
</pre></div>
</div>
<p>The resulting output file can be ingested using <code class="docutils literal"><span class="pre">higlass-manage</span></code>:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>higlass-manage.py ingest --filetype multivec --datatype multivec data.mv5
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">HiGlass</a></h1>








<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="tutorial.html">Tutorial</a><ul>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#prerequisites">Prerequisites</a></li>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#quickly-viewing-a-dataset">Quickly viewing a dataset</a></li>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#running-higlass-locally">Running HiGlass locally</a></li>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#adding-data">Adding data</a></li>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#viewing-data-in-the-client">Viewing data in the client</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="jupyter.html">Jupyter Notebooks</a><ul>
<li class="toctree-l2"><a class="reference internal" href="jupyter.html#installation">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="jupyter.html#usage">Usage</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="python_api.html">Python API</a><ul>
<li class="toctree-l2"><a class="reference internal" href="python_api.html#hgflask">hgflask</a></li>
<li class="toctree-l2"><a class="reference internal" href="python_api.html#hgflask-client">hgflask.client</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="javascript_api.html">Javascript API</a><ul>
<li class="toctree-l2"><a class="reference internal" href="javascript_api.html#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="javascript_api.html#api-functions">API Functions</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Data Preparation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#bed-files">Bed Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#bedpe-like-files">Bedpe-like Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#bedgraph-files">BedGraph files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#bigwig-files">bigWig files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gene-annotation-tracks">Gene Annotation Tracks</a></li>
<li class="toctree-l2"><a class="reference internal" href="#hitile-files">Hitile files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#cooler-files">Cooler files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#multivec-files">Multivec Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#epilogos-data-multivec">Epilogos Data (multivec)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#other-data-multivec">Other Data (multivec)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="track_types.html">Track Types</a><ul>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#linking-value-scales">Linking value scales</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#bed-like">Bed-like</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#gene-annotations">Gene Annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#heatmap">Heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#d-heatmap">1D Heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#d-rectangle-domain">2D Rectangle Domain</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#horizontal-2d-rectangle-domain">Horizontal 2D Rectangle Domain</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#line">Line</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#bar">Bar</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#point">Point</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#id5">1D Heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#chromosome-labels">Chromosome Labels</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#chromosome-grid">Chromosome Grid</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#stacked-bars">Stacked Bars</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#multiple-lines">Multiple Lines</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#multiple-bar-charts">Multiple Bar Charts</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#d-annotations">1D Annotations</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="developer.html">Contributor Guidelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="b_viewer.html">Viewer</a><ul>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#adding-views">Adding views</a></li>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#closing-and-resizing-views">Closing and resizing views</a></li>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#replacing-tracks">Replacing Tracks</a></li>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#synchronizing-zoom-and-location">Synchronizing zoom and location</a></li>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#adding-cross-sections">Adding cross sections</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="higlass_server.html">Server</a><ul>
<li class="toctree-l2"><a class="reference internal" href="higlass_server.html#configuration">Configuration</a></li>
<li class="toctree-l2"><a class="reference internal" href="higlass_server.html#development">Development</a></li>
<li class="toctree-l2"><a class="reference internal" href="higlass_server.html#testing">Testing</a></li>
<li class="toctree-l2"><a class="reference internal" href="higlass_server.html#api">API</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="view_config.html">View Configs</a><ul>
<li class="toctree-l2"><a class="reference internal" href="view_config.html#show-a-specific-genomic-location">Show a specific genomic location</a></li>
<li class="toctree-l2"><a class="reference internal" href="view_config.html#upload-a-viewconf-to-the-server">Upload a viewconf to the server</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="views.html">Views</a><ul>
<li class="toctree-l2"><a class="reference internal" href="views.html#genome-position-search-box">Genome Position Search Box</a></li>
<li class="toctree-l2"><a class="reference internal" href="views.html#view-synchronization">View synchronization</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="javascript_api.html" title="previous chapter">Javascript API</a></li>
      <li>Next: <a href="track_types.html" title="next chapter">Track Types</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2017,2018 HiGlass Authors.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.6.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="_sources/data_preparation.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>