
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Views &#8212; HiGlass v1.0 documentation</title>
    <link rel="stylesheet" href="_static/higlass.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     'v1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="View Configs" href="view_config.html" />
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-77756807-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-77756807-1');
  </script>
  
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />


  </head>
  <body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="views">
<h1>Views<a class="headerlink" href="#views" title="Permalink to this headline">¶</a></h1>
<p>Views are visible units with their own x and y axes. Every track within a view
shares the same view-wide x and y scales. Horizontal tracks share the x axis,
vertical tracks share the y axis and 2D tracks share the x and y. Because 2D
tracks share the same x and y scales, multiple instances of 2D tracks in the
same view must necessarily be overlayed on top of each other. To compare 2D
tracks without overlaying them, they need to be placed in separate views with
synchronized axes.</p>
<div class="section" id="genome-position-search-box">
<h2>Genome Position Search Box<a class="headerlink" href="#genome-position-search-box" title="Permalink to this headline">¶</a></h2>
<img alt="_images/toggle-genome-position-search-box.png" class="align-right" src="_images/toggle-genome-position-search-box.png" />
<p>HiGlass was designed to display genomic data. It arranges this data along a
coordinate system based on genome assemblies. The coordinate system presumes a
particular order of the sequences in an assembly and then displays them in a
linear fashion from end to end. The chromosomal location displayed at any given
moment can be displayed using either a <a class="reference internal" href="track_types.html#chromosome-labels"><span class="std std-ref">chromosome labels tracks</span></a> or the genome position search box.</p>
<img alt="_images/genome-position-search-box.png" class="align-center" src="_images/genome-position-search-box.png" />
<p>The genome position search box, in addition to displaying the current location,
can be used to move the current view to a different location in the genome.
Genomic locations can be entered in a format similar to the one used by the
UCSC genome browser (e.g. <code class="docutils literal"><span class="pre">chr1:10,100,000-20,200,000</span></code>). One can also search
for gene names within the selected assembly.</p>
<p>The genome position search box can be enabled from the view menu.</p>
<div class="section" id="adding-new-assemblies-chromsizes">
<h3>Adding new assemblies (chromSizes)<a class="headerlink" href="#adding-new-assemblies-chromsizes" title="Permalink to this headline">¶</a></h3>
<p>New assemblies have to be loaded into the higlass server in the same manner as
as <a class="reference internal" href="track_types.html#chromosome-labels"><span class="std std-ref">chromosome labels</span></a> with a datatype of <code class="docutils literal"><span class="pre">chromsizes</span></code>
and a filetype of <code class="docutils literal"><span class="pre">chromsizes-tsv</span></code>:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>docker <span class="nb">exec</span> higlass-container python <span class="se">\</span>
    higlass-server/manage.py ingest_tileset <span class="se">\</span>
    --filename /tmp/chromSizes_dm3.tsv <span class="se">\</span>
    --filetype chromsizes-tsv <span class="se">\</span>
    --datatype chromsizes <span class="se">\</span>
    --name dm3
</pre></div>
</div>
<p>The new assembly can chosen from the dropdown in the genome position search box:</p>
<img alt="_images/assembly-selection.png" class="align-center" src="_images/assembly-selection.png" />
</div>
</div>
<div class="section" id="view-synchronization">
<h2>View synchronization<a class="headerlink" href="#view-synchronization" title="Permalink to this headline">¶</a></h2>
<p>To compare data in two or more views, we can synchronize the locations and zoom
levels of multiple different views. Location and zoom synchronization can be
done as an instantaneous on-off operation (“take zoom/location from”) or can
be set as a continuous constraint (“lock zoom/location with”).</p>
<img alt="_images/take_zoom_from.png" class="align-right" src="_images/take_zoom_from.png" />
<div class="section" id="take-zoom-from">
<h3>Take zoom from<a class="headerlink" href="#take-zoom-from" title="Permalink to this headline">¶</a></h3>
<p>Taking the zoom from a different view sets the scaling factor of this view to
that of the target view. Both views remain centered on the same point that they
were centered on before the operation but now they display data at the same
scale.</p>
</div>
<div class="section" id="take-location-from">
<h3>Take location from<a class="headerlink" href="#take-location-from" title="Permalink to this headline">¶</a></h3>
<p>Taking the location from a different view sets the center of this view (along
both the view-wide x and y axes) to the the center of the target view.</p>
</div>
<div class="section" id="take-location-and-zoom-from">
<h3>Take location and zoom from<a class="headerlink" href="#take-location-and-zoom-from" title="Permalink to this headline">¶</a></h3>
<p>Taking the location and zoom from a different view centers and zooms this view
on the same location (e.g. same center point) as the target view.</p>
</div>
<div class="section" id="lock-zoom-with">
<h3>Lock zoom with<a class="headerlink" href="#lock-zoom-with" title="Permalink to this headline">¶</a></h3>
<img alt="_images/lock_zoom_with.png" class="align-right" src="_images/lock_zoom_with.png" />
<p>To compare data at the same scale we may want to link (lock) scales between
views. The “lock zoom with” operation locks two tracks so that they have a
constant difference in scales. If they are the same scale (zoom level) when the
operation is performed, they will always maintain the same scale. If they have
different scales upon locking, then they will always maintain the same ratio
between their scales.</p>
<p>It is possible to link more than two views. The pairwise differences in
parameters are maintained between all of the members of the zoom group.</p>
</div>
<div class="section" id="lock-location-with">
<h3>Lock location with<a class="headerlink" href="#lock-location-with" title="Permalink to this headline">¶</a></h3>
<p>To compare data at the same location we can link (lock) the location between
different views. The “lock location with” operation creates a lock that
maintains a constant difference between the center points of two views. If the
views were centered on the same location upon locking, then they will always
show the same location.</p>
</div>
<div class="section" id="lock-zoom-and-location-with">
<h3>Lock zoom and location with<a class="headerlink" href="#lock-zoom-and-location-with" title="Permalink to this headline">¶</a></h3>
<p>The most common operation is to lock both the location and zoom. This ensures
that two views show the same location and the same scale, making it easy to
compare different experimental conditions.</p>
</div>
<div class="section" id="unlinking">
<h3>Unlinking<a class="headerlink" href="#unlinking" title="Permalink to this headline">¶</a></h3>
<p>Both zoom and location locks can be removed using the view menu.</p>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">HiGlass</a></h1>








<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="tutorial.html">Tutorial</a><ul>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#prerequisites">Prerequisites</a></li>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#quickly-viewing-a-dataset">Quickly viewing a dataset</a></li>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#running-higlass-locally">Running HiGlass locally</a></li>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#adding-data">Adding data</a></li>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#viewing-data-in-the-client">Viewing data in the client</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="jupyter.html">Jupyter Notebooks</a><ul>
<li class="toctree-l2"><a class="reference internal" href="jupyter.html#installation">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="jupyter.html#usage">Usage</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="python_api.html">Python API</a><ul>
<li class="toctree-l2"><a class="reference internal" href="python_api.html#hgflask">hgflask</a></li>
<li class="toctree-l2"><a class="reference internal" href="python_api.html#hgflask-client">hgflask.client</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="javascript_api.html">Javascript API</a><ul>
<li class="toctree-l2"><a class="reference internal" href="javascript_api.html#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="javascript_api.html#api-functions">API Functions</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="data_preparation.html">Data Preparation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#bed-files">Bed Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#bedpe-like-files">Bedpe-like Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#bedgraph-files">BedGraph files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#bigwig-files">bigWig files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#gene-annotation-tracks">Gene Annotation Tracks</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#hitile-files">Hitile files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#cooler-files">Cooler files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#multivec-files">Multivec Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#epilogos-data-multivec">Epilogos Data (multivec)</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#other-data-multivec">Other Data (multivec)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="track_types.html">Track Types</a><ul>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#linking-value-scales">Linking value scales</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#bed-like">Bed-like</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#gene-annotations">Gene Annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#heatmap">Heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#d-heatmap">1D Heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#d-rectangle-domain">2D Rectangle Domain</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#horizontal-2d-rectangle-domain">Horizontal 2D Rectangle Domain</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#line">Line</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#bar">Bar</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#point">Point</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#id5">1D Heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#chromosome-labels">Chromosome Labels</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#chromosome-grid">Chromosome Grid</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#stacked-bars">Stacked Bars</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#multiple-lines">Multiple Lines</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#multiple-bar-charts">Multiple Bar Charts</a></li>
<li class="toctree-l2"><a class="reference internal" href="track_types.html#d-annotations">1D Annotations</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="developer.html">Contributor Guidelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="b_viewer.html">Viewer</a><ul>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#adding-views">Adding views</a></li>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#closing-and-resizing-views">Closing and resizing views</a></li>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#replacing-tracks">Replacing Tracks</a></li>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#synchronizing-zoom-and-location">Synchronizing zoom and location</a></li>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#adding-cross-sections">Adding cross sections</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="higlass_server.html">Server</a><ul>
<li class="toctree-l2"><a class="reference internal" href="higlass_server.html#configuration">Configuration</a></li>
<li class="toctree-l2"><a class="reference internal" href="higlass_server.html#development">Development</a></li>
<li class="toctree-l2"><a class="reference internal" href="higlass_server.html#testing">Testing</a></li>
<li class="toctree-l2"><a class="reference internal" href="higlass_server.html#api">API</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="view_config.html">View Configs</a><ul>
<li class="toctree-l2"><a class="reference internal" href="view_config.html#show-a-specific-genomic-location">Show a specific genomic location</a></li>
<li class="toctree-l2"><a class="reference internal" href="view_config.html#upload-a-viewconf-to-the-server">Upload a viewconf to the server</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Views</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#genome-position-search-box">Genome Position Search Box</a></li>
<li class="toctree-l2"><a class="reference internal" href="#view-synchronization">View synchronization</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="view_config.html" title="previous chapter">View Configs</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2017,2018 HiGlass Authors.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.6.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="_sources/views.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>